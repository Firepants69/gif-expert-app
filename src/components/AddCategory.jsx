import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  
  const [inputValue, setInputValue] = useState("");

  const onInputValue = ({ target }) => {
    setInputValue(target.value);
  };


  const onSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();

    if (newInputValue.length < 1) return;

    onNewCategory(newInputValue); 
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="BuscarGif"
        value={inputValue}
        onChange={onInputValue}
      />
    <button className="submit-button-category" type="submit">aceptar</button>
    </form>
  );
};