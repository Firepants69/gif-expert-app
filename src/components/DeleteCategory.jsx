import '../styles.css'

export const DeleteCategory = ({onDeleteCategory,category})=>{
    return(
        <button className="DeleteCategoryButton" onClick={()=>{onDeleteCategory(category)}}>x</button>
    )
}