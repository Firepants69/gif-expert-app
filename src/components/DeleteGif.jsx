import { useState } from "react";
import '../styles.css'

export const DeleteGif = ({id,setImages})=>{
    
    const newImages = (images)=>{
        let newImagesArray = []
            for(let image of images){
                if(image.id != id){
                    newImagesArray.push(image)
                }
            }
            return newImagesArray
    }
    const deleteGif = ()=>{
        setImages((images)=>newImages(images)    
        )
    
    }
    
    return(
        < button  className="DeleteGifBoton" onClick={
            deleteGif
        }>x</button>
    )
}