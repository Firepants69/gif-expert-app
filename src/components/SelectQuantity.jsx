
import React from 'react'

export const SelectQuantity = ({limit,OnLimitChange}) => {
  
 
  return (
    <>
      <input type="range" className='range-limit' max={50} min={1} onChange={OnLimitChange}/>
        <p>cantidad de gifs: <span className='limit-count'>{limit}</span></p>
            
    </>
  )
}

