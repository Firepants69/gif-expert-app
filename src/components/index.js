export * from './AddCategory'
export * from './DeleteCategory'
export * from './DeleteGif'
export * from './GifItem'
export * from './GifGrid'