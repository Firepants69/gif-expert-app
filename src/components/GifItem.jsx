import { useState } from "react";
import { DeleteGif } from "./DeleteGif";
import PropTypes from 'prop-types'

export const GifItem = ({ title, url,id,setImages,onDeleteCategory,category}) => {
    return (
        <div className="card">
            <DeleteGif id={id} setImages={setImages}onDeleteCategory={onDeleteCategory}category={category}/>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.PropTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}