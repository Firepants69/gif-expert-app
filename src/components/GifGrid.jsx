
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import { DeleteCategory } from "./DeleteCategory";

export const GifGrind = ({ category,onDeleteCategory,Initiallimit }) => {
    
    const { images,setImages, isLoading} = useFetchGifs(category,Initiallimit)

    

    return (
        <>
            <h3>{category} <DeleteCategory onDeleteCategory={onDeleteCategory}category={category}/></h3>
            {
                isLoading ? <h2>cargando...</h2> : null
            }
         
            <div className="card-grid">
              
                {images.map((image) => (
                    <GifItem
                        key={image.id}
                        title ={image.title}
                        url={image.url}
                        id={image.id}
                        setImages = {setImages}
                        onDeleteCategory={onDeleteCategory}
                        category={category}
                    ></GifItem>
                ))}
            </div>
        </>
    )
}