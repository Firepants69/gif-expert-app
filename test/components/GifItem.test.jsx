import { render,screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('testeo en GifItem', () => {

  const title = 'Saitama'
  const url = 'https://one-punch.com/saitama.jpg'
  

  test('debe de hacer match con el snapshot', () => {

    const {container} =render(<GifItem title={title} url={url}></GifItem>)
    expect(container).toMatchSnapshot(); 

  })
  test('debe de mostrar la imagen con el URL indicado', () => {
    
    render(<GifItem title={title} url={url}></GifItem>)
    // screen.debug()
    const {src,alt} = screen.getByRole('img')
    expect(src).toBe(url);
    expect(alt).toBe(title);
    
  })

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url}></GifItem>) 
    expect(screen.getByText(title)).toBeTruthy();
  })
  
  
  
})
