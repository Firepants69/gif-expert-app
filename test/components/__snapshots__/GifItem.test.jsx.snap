// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testeo en GifItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <button
      class="DeleteGifBoton"
    >
      x
    </button>
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
